pkg/tasks/tasks.go

```diff
@@ -40,6 +40,15 @@ func (t *Task) addAction(action taskAction) {
 	t.actions = append(t.actions, action)
 }
 
+func (t *Task) addActionWithBuilder(desc string, runFunc actionRunFunc) *actionWithBuilder {
```

Comment:
Since there's no `addAction` without a builder I would prefer omitting the `WithBuilder` I know it's more explicit but I don't think we care in our case.

Reply:
`addAction` is 4 lines above :-D